src/routers/alpha-router/alpha-router.ts

```diff
@@ -2175,58 +2230,60 @@ export class AlphaRouter
       );
     }
 
-    // Maybe Quote mixed routes
-    // if MixedProtocol is specified or no protocol is specified and v2 is supported AND tradeType is ExactIn
-    // AND is Mainnet or Gorli
-    if (shouldQueryMixedProtocol && mixedProtocolAllowed) {
-      log.info({ protocols, tradeType }, 'Routing across MixedRoutes');
+    if (!fotInDirectSwap) {
+      // Maybe Quote mixed routes
+      // if MixedProtocol is specified or no protocol is specified and v2 is supported AND tradeType is ExactIn
+      // AND is Mainnet or Gorli
+      if (shouldQueryMixedProtocol && mixedProtocolAllowed) {
```

Comment:
shall you add it in this if statement?+

Reply:
I think it's more readable to make a outer-scoped if statement for `fotInDirectSwap`.  Also there are some complex conditionals that makes it prone to make coding mistakes, e.g. 

```
if (
        v3ProtocolSpecified ||
        noProtocolsSpecified ||
        (shouldQueryMixedProtocol && mixedProtocolAllowed)
   ) {}
```

I can make it such that:

```
if (
        (v3ProtocolSpecified ||
        noProtocolsSpecified ||
        (shouldQueryMixedProtocol && mixedProtocolAllowed)) &&
       fotInDirectSwap
   ) {}

```

But it's more readable with:

```
if (!fotInDirectSwap) {
      if (
        v3ProtocolSpecified ||
        noProtocolsSpecified ||
        (shouldQueryMixedProtocol && mixedProtocolAllowed)
      ) {
       }
}
```

Reply:
fair enough